includes/admin/upgrades/upgrade-functions.php

```diff
@@ -1328,499 +1328,294 @@ function edd_include_file_download_log_migration_batch_processor( $class = '' )
+	</p>
 
-					if ( next_step_wrapper.hasClass( 'metabox-holder' ) ) {
-						if ( next_step_wrapper.find( '.edd-migration.allowed' ).length ) {
-							next_step_wrapper.find('.edd-migration.allowed').show();
-							next_step_wrapper.find('.edd-migration.unavailable').hide();
+	<p>
+		<?php printf( esc_html__( 'This migration can also be run via WP-CLI with the following command: %s. This is the recommended method for large sites.', 'easy-digital-downloads' ), '<code>wp edd v30_migration</code>' ); ?>
+	</p>
 
-							if (auto_start_next_step) {
-								next_step_wrapper.find('.edd-export-form:not(#edd-remove-legacy-data-form)').submit();
-							}
+	<?php
+	// Only show the migration form if there are still upgrades to do.
+	if ( ! $can_remove_legacy_data ) : ?>
+	<form id="edd-v3-migration" class="edd-v3-migration" method="POST">
+		<p>
+			<label for="edd-v3-migration-confirmation">
+				<input type="checkbox" id="edd-v3-migration-confirmation" class="edd-v3-migration-confirmation" name="backup_confirmation" value="1">
+				<?php esc_html_e( 'I have secured a backup of my website data.', 'easy-digital-downloads' ); ?>
+			</label>
+		</p>
+		<input type="hidden" name="_wpnonce" value="<?php echo esc_attr( wp_create_nonce( 'edd_process_v3_upgrade' ) ); ?>">
+		<button type="submit" id="edd-v3-migration-button" class="button button-primary disabled" disabled>
+			<?php esc_html_e( 'Upgrade Easy Digital Downloads', 'easy-digital-downloads' ); ?>
+		</button>
+		<div class="edd-v3-migration-error edd-hidden"></div>
+	</form>
+	<?php endif
+
+	/*
+	 * Progress is only shown immediately if the upgrade is in progress. Otherwise it's hidden by default
+	 * and only revealed via JavaScript after the process has started.
+	 */
+	?>
+	<div id="edd-migration-progress" <?php echo count( array_filter( $upgrade_statuses ) ) ? '' : 'class="edd-hidden"'; ?>>
+		<ul>
+			<?php foreach ( $upgrades as $upgrade_key => $upgrade_details ) :
+				// We skip the one to remove legacy data. We'll handle that separately later.
+				if ( 'v30_legacy_data_removed' === $upgrade_key ) {
+					continue;
+				}
+				?>
+				<li id="edd-v3-migration-<?php echo esc_attr( sanitize_html_class( $upgrade_key ) ); ?>" <?php echo true === $upgrade_statuses[ $upgrade_key ] ? 'class="edd-upgrade-complete"' : ''; ?> data-upgrade="<?php echo esc_attr( $upgrade_key ); ?>">
+					<span class="edd-migration-status">
+						<?php
+						if ( true === $upgrade_statuses[ $upgrade_key ] ) {
+							?>
+							<span class="dashicons dashicons-yes"></span>
+							<span class="screen-reader-text"><?php esc_html_e( 'Upgrade complete', 'easy-digital-downloads' ); ?></span>
+							<?php
 						} else {
-							processNext( next_step_wrapper );
+							?>
+							<span class="dashicons dashicons-ellipsis"></span>
+							<span class="screen-reader-text"><?php esc_html_e( 'Upgrade pending', 'easy-digital-downloads' ); ?></span>
```

Comment:
This text should change once the upgrade has completed, I think.

Wondering if it would be possible to have it set initially to "pending" or "waiting", and then "in progress", and then "complete"?

OR...wondering if this is necessary at all, since the percentage is read by screen readers...so it could just be removed?

Reply:
Noting that if I reload the page, the percentage is gone, and this span does read "Upgrade complete", so maybe we just need to adjust the script behavior a bit.

Reply:
I've pushed up a change that updates the screen reader text when the step completes.

The percentage is gone when you reload the page because I'm not going to make that work perfectly until I focus on step recovery. I'm going to do that in a separate issue once this PR gets merged.